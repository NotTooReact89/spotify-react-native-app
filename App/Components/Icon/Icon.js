// flow

import React from 'react'
import { Image } from 'react-native'
import Images from '../../Images'
import { TestProps } from '../../Lib'
import Styles, { iconSize } from './IconStyles'

export type IconType = $Keys<typeof Images.icons>

export type IconProps = {
  type: IconType,
  size?: number,
  testId?: string,
  style?: any,
}

const Icon = ({ type, size, testId, style }: IconProps) => {
  const testProps = testId ? TestProps(testId) : null
  return (
    <Image
      source={Images.icons[type]}
      style={[Styles.icon, { height: size, width: size }, style]}
      {...testProps}
    />
  )
}

Icon.defaultProps = {
  testId: '',
  size: iconSize,
  style: {},
}

export default Icon
