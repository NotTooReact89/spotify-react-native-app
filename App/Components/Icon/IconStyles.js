// @flow

import { StyleSheet } from 'react-native'

export const iconSize = 24

export default StyleSheet.create({
  icon: {
    resizeMode: 'contain',
    alignItems: 'center',
    justifyContent: 'center',
  },
})
