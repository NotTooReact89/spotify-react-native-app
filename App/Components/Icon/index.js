// @flow

export { iconSize } from './IconStyles'
export { default, IconType } from './Icon'
