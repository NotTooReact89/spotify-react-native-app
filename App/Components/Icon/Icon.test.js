// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Icon from './Icon'

it('should render an icon', () => {
  const tree = renderer
    .create(<Icon type="tickInverse" testId="someId" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render an icon with custom size', () => {
  const tree = renderer
    .create(<Icon type="tickInverse" size={14} testId="someId" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render an icon with custom style', () => {
  const tree = renderer
    .create(
      <Icon
        type="tickInverse"
        style={{ backgroundColor: 'purple' }}
        testId="someId"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
