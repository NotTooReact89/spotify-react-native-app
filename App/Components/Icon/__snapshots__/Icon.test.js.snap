// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an icon 1`] = `
<Image
  source={
    Object {
      "process": [Function],
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "resizeMode": "contain",
      },
      Object {
        "height": 24,
        "width": 24,
      },
      Object {},
    ]
  }
  testID="someId"
/>
`;

exports[`should render an icon with custom size 1`] = `
<Image
  source={
    Object {
      "process": [Function],
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "resizeMode": "contain",
      },
      Object {
        "height": 14,
        "width": 14,
      },
      Object {},
    ]
  }
  testID="someId"
/>
`;

exports[`should render an icon with custom style 1`] = `
<Image
  source={
    Object {
      "process": [Function],
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "resizeMode": "contain",
      },
      Object {
        "height": 24,
        "width": 24,
      },
      Object {
        "backgroundColor": "purple",
      },
    ]
  }
  testID="someId"
/>
`;
